src/config_parser/parser.rs

```diff
@@ -52,4 +52,38 @@ impl Config {
             })
         })
     }
+    /// A helper function which returns an appropriate config file path checking if the config
+    /// file exists on that path.
+    ///
+    /// # Error
+    ///
+    /// Returns a `config file not found!!` error if the config file is not present under following
+    /// paths which are:
+    /// 1. `~/.config/websurfx/` if it not present here then it fallbacks to the next one (2)
+    /// 2. `/etc/xdg/websurfx/config.lua` if it is not present here then it fallbacks to the next
+    ///    one (3).
+    /// 3. `websurfx/` (under project folder ( or codebase in other words)) if it is not present
+    ///    here then it returns an error as mentioned above.
+    fn handle_different_config_file_path() -> Result<String, Box<dyn std::error::Error>> {
+        if Path::new(
+            format!(
+                "{}/.config/websurfx/config.lua",
+                std::env::var("HOME").unwrap()
+            )
+            .as_str(),
+        )
+        .exists()
+        {
+            Ok(format!(
+                "{}/.config/websurfx/config.lua",
+                std::env::var("HOME").unwrap()
+            ))
+        } else if Path::new("/etc/xdg/websurfx/config.lua").exists() {
+            Ok("/etc/xdg/websurfx/config.lua".to_string())
+        } else if Path::new("./websurfx/config.lua").exists() {
+            Ok("./websurfx/config.lua".to_string())
```

Comment:
I noticed that `websurfx/config.lua` is repeated multiple times in the code. This could potentially lead to errors, particularly if we make a typo like `websurrfx/config.lua` somewehre. To avoid such issues, I suggest using a variable to represent `websurfx/config.lua` and utilizing it instead of a raw string.

Reply:
Yes, you are right :+1: :smile: . I will make the change right away, just give me a few seconds :slightly_smiling_face: .